soccer/gameplay/tests/test_field.py

```diff
@@ -0,0 +1,151 @@
+import unittest
+import main
+import robocup
+import evaluation.field
+import constants
+
+class Moc_Robot:
+	def __init__(self, x, y):
+		self.pos = robocup.Point(x, y)
+		self.visible = True;
+
+	def set_pos(self, x, y):
+		self.pos = robocup.Point(x, y)
+
+class TestField(unittest.TestCase):
+	def __init__(self, *args, **kwargs):
+		super(TestField, self).__init__(*args, **kwargs)
+
+	def setUp(self):
+		self.robots = [Moc_Robot(0, 0), Moc_Robot(0, 0), Moc_Robot(0, 0), 
+				Moc_Robot(0, 0), Moc_Robot(0, 0), Moc_Robot(0, 0)]
+		main.set_their_robots(self.robots)
+
+	# Set 6 robots position to a single point
+	#
+	# @param x final x coordinate of robots
+	# @param y final y coordinate of robots
+	#
+	def set_robot_pos(self, x, y):
+		for robot in self.robots:
+			robot.set_pos(x, y)
+		main.set_their_robots(self.robots)
+
+	# Positions 6 robots around a point but not on it
+	#
+	# @param x: x coordinate of point
+	# @param y: y coordinate of point
+	# 
+	def set_robots_around_pos(self, x, y, rad):
+		self.robots[0].set_pos(x - constants.Robot.Radius * rad, y - constants.Robot.Radius * rad)
+		self.robots[1].set_pos(x - constants.Robot.Radius * rad, y + constants.Robot.Radius * rad)
+		self.robots[2].set_pos(x, y + constants.Robot.Radius * rad)
+		self.robots[3].set_pos(x + constants.Robot.Radius * rad, y - constants.Robot.Radius * rad)
+		self.robots[4].set_pos(x + constants.Robot.Radius * rad, y)
+		self.robots[5].set_pos(x, y - constants.Robot.Radius * rad)
+		main.set_their_robots(self.robots)
+
+	def test_space_coeff_at_pos(self):
+		width = constants.Field.Width
+		length = constants.Field.Length
+
+		msg = "evaluation.field.space_coeff_at_pos not functioning as intended: "
+
+		# Easier than writing out whole function
+		# Finds space_coeff_at_pos for a point
+		#
+		# @param x: x coordinate of point
+		# @param y: y coordinate of point
+		def run_function(x, y):
+			return evaluation.field.space_coeff_at_pos(robocup.Point(x, y))
+		
+		# both robots and position are at (0, 0)
+		self.assertTrue(run_function(0, 0) == 1, msg = msg + "Position occupied by robots is considered open")
```

Comment:
You shouldn't need to add the `msg = msg +` part. The output of a failed test already tells us which function isn't working. The `msg` argument should only be there to describe why that test failed. Here is the output for just a `assertTrue(False, 'failure message goes here')`

```
test_fail (__main__.FailureMessageTest) ... FAIL

======================================================================
FAIL: test_fail (__main__.FailureMessageTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_failwithmessage.py", line 6, in test_fail
    self.assertTrue(False, 'failure message goes here')
AssertionError: failure message goes here

----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (failures=1)
```

The `"Position occupied by robots is considered open"` should be more than enough to explain the failure.